pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -2155,22 +2180,34 @@ async def _get_event_iterator(self) -> AsyncIterator[ModelResponseStreamEvent]:
                 pass  # there's nothing we need to do here
 
             elif isinstance(chunk, responses.ResponseReasoningSummaryTextDeltaEvent):
-                yield self._parts_manager.handle_thinking_delta(
+                for event in self._parts_manager.handle_thinking_delta(
                     vendor_part_id=f'{chunk.item_id}-{chunk.summary_index}',
                     content=chunk.delta,
                     id=chunk.item_id,
-                )
+                ):
+                    yield event
+
+            elif isinstance(chunk, responses.ResponseReasoningTextDeltaEvent):
+                for event in self._parts_manager.handle_thinking_delta(
+                    vendor_part_id=chunk.item_id,
+                    raw_content_delta=chunk.delta,
+                    raw_content_index=chunk.content_index,
```

Comment:
See notes on parts manager, we have to handle this differently. I imagine something like:

```py
def update_provider_details(details: dict[str, Any] | None) -> dict[str, Any]:
    raw_content = (details or {}).get('raw_content')
    # here figure out if raw_content is a list,
    # whether the index already has a string,
    # and then add the delta
    
    return {'raw_content': raw_content}

self._parts_manager.handle_thinking_delta(..., provider_details=update_provider_details)
```

That `update_provider_details` could then be stored on `ThinkingPartDelta` and applied to the `ThinkingPart` in `ThinkingPartDelta.apply`